src/main/scala/treadle/executable/ExecutionEngine.scala

```diff
@@ -215,10 +228,20 @@ class ExecutionEngine(
         }
         println(s"${symbol.name} <= $value")
       }
+      val currentValue = dataStore(symbol)
       dataStore.update(symbol, adjustedValue)
       vcdOption.foreach { vcd =>
         vcd.wireChanged(symbol.name, dataStore(symbol), symbol.bitWidth)
       }
+
+      if(
+        scheduler.triggeredAssigns.contains(symbol) &&
+        currentValue == Big0 && adjustedValue == Big1
+      ) {
+
+        scheduler.executeCombinationalAssigns()
+        scheduler.executeTriggeredAssigns(symbol)
```

Comment:
Is this necessary? If `scheduler.executeCombinationalAssigns()` doesn't clear `executeTriggeredAssigns`, doesn't the previous logic guarantee that `scheduler.triggeredAssigns.contains(symbol)`?
Perhaps I need to look more closely at `executeTriggeredAssigns` and the conditions under which it is cleared.

Reply:
triggeredAssigns are not cleared, they are simply a list of clock or clock like things where a particular transition could change registers or other downstream logic.  executeCombinational goes first just to make sure that any other inputs or things hanging around are up to date to before the triggeredAssigns code, e.g. that the register input is there before the clock latches that value into the register output. It is possible that there is some over freshening of the circuit, but the current state of treadle is a fairly complex  balance between trying not to run unnecessary evaluation and not missing anything any calculation that happens because of derived clocks etc.